exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SQLExpressionBuilder.kt

```diff
@@ -96,6 +93,37 @@ fun <T : Any?> ExpressionWithColumnType<T>.varPop(scale: Int = 2): VarPop<T> = V
  */
 fun <T : Any?> ExpressionWithColumnType<T>.varSamp(scale: Int = 2): VarSamp<T> = VarSamp(this, scale)
 
+// Array Comparisons
+
+/** Returns this subquery wrapped in the `ANY` operator. This function is not supported by the SQLite dialect. */
+fun <T> anyFrom(subQuery: Query): Op<T> = AllAnyFromSubQueryOp(true, subQuery)
+
+/** Returns this array of data wrapped in the `ANY` operator. This function is only supported by PostgreSQL and H2 dialects. */
+fun <T> anyFrom(array: Array<T>): Op<T> = AllAnyFromArrayOp(true, array)
+
+/** Returns this table wrapped in the `ANY` operator. This function is only supported by PostgreSQL and H2 dialects. */
+fun <T> anyFrom(table: Table): Op<T> = AllAnyFromTableOp(true, table)
+
+/** Returns this subquery wrapped in the `ALL` operator. This function is not supported by the SQLite dialect. */
+fun <T> allFrom(subQuery: Query): Op<T> = AllAnyFromSubQueryOp(false, subQuery)
+
+/** Returns this array of data wrapped in the `ALL` operator. This function is only supported by PostgreSQL and H2 dialects. */
+fun <T> allFrom(array: Array<T>): Op<T> = AllAnyFromArrayOp(false, array)
+
+/** Returns this table wrapped in the `ALL` operator. This function is only supported by PostgreSQL and H2 dialects. */
+fun <T> allFrom(table: Table): Op<T> = AllAnyFromTableOp(false, table)
+
+// TODO Currently these functions delegate to `anyFrom`. Should the actual `SOME` SQL operator be supported?
+
+/** An alias for [anyFrom]. */
+fun <T> someFrom(subQuery: Query): Op<T> = anyFrom(subQuery)
+
+/** An alias for [anyFrom]. */
+fun <T> someFrom(array: Array<T>): Op<T> = anyFrom(array)
+
+/** An alias for [anyFrom]. */
+fun <T> someFrom(table: Table): Op<T> = anyFrom(table)
+
```

Comment:
@ShreckYe There's already quite a bit in this PR. Please remove these aliases and then `SOME` can be introduced in another PR once this one is finalized. It won't take much tweaking with the current setup.